lesson_18/homework_18.py

```diff
@@ -0,0 +1,88 @@
+"""
+Є вiдкритий API NASA який дозволяє за певними параметрами отримати данi 
+у виглядi JSON про фото зробленi ровером “Curiosity” на Марсi. Серед цих 
+даних є посилання на фото якi потрiбно розпарсити i потiм за допомогою 
+додаткових запитiв скачати i зберiгти цi фото як локальнi файли mars_photo1.jpg , 
+mars_photo2.jpg . Завдання потрiбно зробити використовуючи модуль requests
+"""
+import json
+import requests
+from pathlib import Path
+from requests.exceptions import HTTPError, ConnectionError, Timeout
+
+
+def get_data_in_json(url, params):
+    """ Getting data... """
+    try:
+        response = requests.get(url, params, timeout=30)
+        response.raise_for_status()
+        try:
+            data = response.json()
+            return data
+        except json.JSONDecodeError as e:
+           print('Помилка при серіалізації JSON:', e)
+       
+    except HTTPError as e:
+        print('HTTP Помилка:', e)
+    
+    except ConnectionError as e:
+        print('Помилка з\'єднання:', e)
+    except Timeout as e:
+        print('Часова помилка:', e)
+    except Exception as e:
+        print('Невідома помилка:', e)
+
+"""{'photos': 
+    [
+        {
+         'id': 102693, 
+         'sol': 1000,
+         'camera': {'id': 20, 'name': 'FHAZ', 'rover_id': 5, 'full_name': 'Front Hazard Avoidance Camera'}, 
+         'img_src': 'http://mars.jpl.nasa.gov/msl-raw-images/proj/msl/redops/ods/surface/sol/01000/opgs/edr/fcam/FLB_486265257EDR_F0481570FHAZ00323M_.JPG', 
+         'earth_date': '2015-05-30', 
+         'rover': {'id': 5, 'name': 'Curiosity', 'landing_date': '2012-08-06', 'launch_date': '2011-11-26', 'status': 'active'}
+         },
+          
+         {'id': 102694, 'sol': 1000, 
+          'camera': {'id': 20, 'name': 'FHAZ', 'rover_id': 5, 'full_name': 'Front Hazard Avoidance Camera'},
+           'img_src': 'http://mars.jpl.nasa.gov/msl-raw-images/proj/msl/redops/ods/surface/sol/01000/opgs/edr/fcam/FRB_486265257EDR_F0481570FHAZ00323M_.JPG', 
+           'earth_date': '2015-05-30', 
+           'rover': {'id': 5, 'name': 'Curiosity', 'landing_date': '2012-08-06', 'launch_date': '2011-11-26', 'status': 'active'}
+           }
+    ]
+}"""
+
+def save_pic(json_data):
+    
+    directory = Path(r'C:\Users\dmytr\OneDrive\Desktop\LYUBA\Hillel\aqa_070225\lesson_18\Mars photos')
```

Comment:
```suggestion
    directory = Path(r'C:\Users\dmytr\OneDrive\Desktop\LYUBA\Hillel\aqa_070225\lesson_18\Mars photos')
```
Абсолютні шляхи -абсолютне зло